File: 3447-clear-digits/3447-clear-digits.go
func isNum(ch rune) bool {
    return ch >= '0' && ch <= '9'
}

func clearDigits(s string) string {
    var stack []rune

    for _, ch := range s {
        if isNum(ch) {
            if len(stack) > 0 {
                stack = stack[:len(stack) - 1]
            }
        } else {
            stack = append(stack, ch)
        }
    }

    return string(stack)
}

File: 3447-clear-digits/3447-clear-digits.cpp
class Solution {
public:
    bool isNum(char ch) { return (ch >= '0') && (ch <= '9'); }

    string clearDigits(string s) {
        stack<char> st;
        string res = "";

        for (const char& ch : s) {
            if (isNum(ch)) {
                if (!st.empty())
                    st.pop();
            } else {
                st.push(ch);
            }
        }

        while (!st.empty()) {
            res.push_back(st.top());
            st.pop();
        }

        reverse(res.begin(), res.end());

        return res;
    }
};
